File: cs/exts/assetimport_unity/Importer/EnvironmentHDRI.cs
#if UNITY_EDITOR

namespace AssetExchange.Importer
{
    using System;
    using System.IO;
    using System.Linq;
    using AssetExchange.LightJSON;
    using UnityEditor;
    using UnityEngine;

    public static class EnvironmentHDRI
    {
        public static void Import(JsonObject asset, JsonObject selectedVariants)
        {
            // explode variants
            var (variantLabels, variantConfigs) = AssetUtils.VariantUtil.Explode("Primary", selectedVariants);

            // iterate variant config
            foreach (var variantConfig in variantConfigs)
            {
                // get environment map
                var objectList = AssetUtils.VariantUtil.FilterObjectsByVariantConfig(asset, "Primary", variantLabels, variantConfig);
                if (objectList.Count == 0)
                    return;
                var envMap = objectList[0];

                // target paths
                var relTargetPath = "Assets/Import/" + asset["sourceUid"].AsString;
                var targetPath = Path.Combine(Application.dataPath, "Import", asset["sourceUid"].AsString);

                Directory.CreateDirectory(targetPath);

                // find free name
                var texName = AssetUtils.PathUtil.FreeDirEntryName(
                    targetPath,
                    asset["assetId"].AsString.Replace('.', '_') + "_" + string.Join("_", variantConfig.Select(i => i.AsString)),
                    Path.GetExtension(envMap["file"]["path"])
                );

                // import equirectangular hdri
                File.Copy(envMap["file"]["path"], Path.Combine(targetPath, texName));
                AssetDatabase.Refresh();

                // convert to cubemap
                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(relTargetPath + "/" + texName);
                importer.maxTextureSize = 8192;
                importer.textureShape = TextureImporterShape.TextureCube;
                importer.SaveAndReimport();

                // load cubemap
                var cubemap = (Cubemap)AssetDatabase.LoadAssetAtPath(relTargetPath + "/" + texName, typeof(Cubemap));
                if (cubemap == null)
                {
                    throw new Exception("Could not load " + texName);
                }

                // create material
                var matName = Path.GetFileNameWithoutExtension(texName) + ".mat";
                var mat = new Material(Shader.Find("Skybox/Cubemap"));
                mat.SetTexture("_Tex", cubemap);
                AssetDatabase.CreateAsset(mat, relTargetPath + "/" + matName);

                // activate environment map
                RenderSettings.skybox = mat;
            }
        }
    }

}

#endif

File: py/exts/assetimport_blender/importer/environment_hdri.py
import bpy
import assetexchange_shared

def environment_hdri(asset, selectedVariants):
    # explode variants
    variantLabels, variantConfigs = assetexchange_shared.asset.explode_variants('Primary', selectedVariants)

    # iterate variant config
    for variantConfig in variantConfigs:

        # get environment map
        object_list = assetexchange_shared.asset.filter_objects_by_variant_config(asset, 'Primary', variantLabels, variantConfig)
        if len(object_list) == 0:
            return
        env_map = object_list[0]

        # enable world nodes
        world = bpy.data.worlds['World']
        world.use_nodes = True
        nodes = world.node_tree.nodes
        
        # create texture node
        env_text_node = nodes.new('ShaderNodeTexEnvironment')
        env_text_node.image = bpy.data.images.load(env_map["file"]["path"])
        env_text_node.show_texture = True
        env_text_node.image.colorspace_settings.name = "Linear"

        # link to world
        world.node_tree.links.new(
            nodes.get("Background").inputs['Color'], env_text_node.outputs['Color'])
